src/net_processing.cpp

```diff
@@ -759,7 +759,8 @@ void Misbehaving(NodeId pnode, int howmuch)
 static bool StaleBlockRequestAllowed(const CBlockIndex* pindex, const Consensus::Params& consensusParams)
 {
     AssertLockHeld(cs_main);
-    return (pindexBestHeader != nullptr) &&
+    if (chainActive.Contains(pindex)) return true;
```

Comment:
Now that this function is expected to accept main chain blocks, maybe change the function name to `BlockRequestAllowed`?